### ISSUE TITLE
join on multiple conditions

### ISSUE THREAD

**Issue by georgi-yakimov** [nhibernate/nhibernate-core#2330]
hi,

I've run into the following problem when trying to do a join on multiple columns:

```sql
select Id, col, fk_t2
from dbo.table1 t1
left outer join dbo.table2 t2 on t1.fk_t2 = t2.Id and t2.col = t1.col
where <filtering>
```

but the output is:

```sql
select Id, col, fk_t2
from dbo.table1 t1
left outer join dbo.table2 t2 on t1.fk_t2 = t2.Id and t2.col = t1_1.col
where <filtering>
```

The problem is that when there's inheritance join with a withClause and the column which is used in the withClause is present in the base and the inherited table the condition is not generated correctly. The table that is considered in the with clause is from the base table which is not yet joined.

I've checked the code and I think I've found a solution. Can you give me privileges to create a topic branch?

**Comment by gliljas** [comment]
You should fork and create a PR.
https://github.com/nhibernate/nhibernate-core/blob/master/CONTRIBUTING.md

**Comment by fredericDelaporte** [comment]
Fixed by #2361 and #2545.
